### PR TITLE
Faster events

### PR DESCRIPTION
**Pull request by reginafcompton**
This PR addresses the first checkbox discussed here: https://github.com/datamade/la-metro-councilmatic/issues/403#issuecomment-465208479

I plan to move through issue #403 iteratively, taking on each checkbox, one at a time.

### CODE REVIEW

File: lametro/models.py [hunk 1]
```diff
@@ -417,23 +417,6 @@ def media(self):
         return query_set
 
 
-class LAMetroEventMedia(EventMedia):
```
Comment (reginafcompton) [inline lametro/models.py]: EventMedia queried from `prefetch_related` are not the child class (LAMetroEventMedia). [Apparently, that not yet possible in Django.](85aab34288b5e51fb90031cbaf9616fcfea53ae1) 

So, I moved this to a template filter (for now).

File: lametro/models.py [hunk 2]
```diff
@@ -409,31 +409,6 @@ def upcoming_committee_meetings(cls):
         return meetings
 
 
-    @property
-    def media(self):
```
Comment (reginafcompton) [inline lametro/models.py]: Replacing this with a template filter that sorts the media_urls set in the template, without hitting the database, again.

File: lametro/views.py [hunk 3]
```diff
@@ -299,7 +301,9 @@ def get_context_data(self, **kwargs):
 
         # If all meetings
         elif self.request.GET.get('show'):
-            all_events = LAMetroEvent.objects.order_by('-start_time')
+            all_events = LAMetroEvent.objects\
+                                     .order_by('-start_time')\
+                                     .prefetch_related('media_urls')
```
Comment (hancush) [inline lametro/views.py]: do we see the same gains if we use the `Prefetch` object to return `LAMetroEventMedia` objects? this works in the shell:

```python
from django.db.models import Prefetch

LAMetroEvent.objects.filter(SOME FILTER).prefetch_related(Prefetch('media_urls', queryset=LAMetroEventMedia.objects.all()))
```

File: lametro/models.py (hunk 2 above)
Comment (reginafcompton) [reply lametro/models.py]: Update! I am doing the ordering of media_urls in the original queryset.

File: lametro/views.py (hunk 3 above)
Comment (reginafcompton) [reply lametro/views.py]: We do get the same gains! Building off of that....do you think it's too abstruse to order the Event Media when prefetching? I implemented how that would work here:

https://github.com/datamade/la-metro-councilmatic/pull/406/files#diff-803b5091212eac625d7bc1c2dd933c21R322

**Comment by reginafcompton** [comment]
@hancush - ready for additional review!

File: lametro/views.py [hunk 4]
```diff
@@ -271,8 +283,24 @@ class LAMetroEventsView(EventsView):
...
+
+        We prefetch EventMedia (i.e., 'media_urls') in the Events querysets: 
+        this makes for a more efficient page load. 
+        We also order the 'media_urls' by label, ensuring that links to SAP audio
+        come after links to English audio. 'mediaqueryset' facilitates 
+        the ordering of prefetched 'media_urls'.
+        '''
         context = super(LAMetroEventsView, self).get_context_data(**kwargs)
 
+        mediaqueryset = LAMetroEventMedia.objects.annotate(
```
Comment (hancush) [inline lametro/views.py]: how would you feel about moving this prefetch up into the existing event manager [as an extra method](https://docs.djangoproject.com/en/2.1/topics/db/managers/#adding-extra-manager-methods), e.g. `with_media`? that way we have an easy way to access, without repeating code. 

with that change, you can then get the events with media prefetched, like so:

```python
LAMetroevent.objects.with_media().filter(...)   # etc.
```

as far as i can tell, we still get the gains? (i've been looking at the django debug toolbar – are you monitoring a different way? this was my q from semaphor, btw!)

Comment (reginafcompton) [reply lametro/views.py]: sounds excellent! (yeah, I was not pleased about the level of DRY-ness here.) 

And yes, I am using django debug toolbar!

File: lametro/views.py [hunk 5]
```diff
@@ -130,15 +130,19 @@ def post(self, request, *args, **kwargs):
 
     def get_context_data(self, **kwargs):
         context = super(EventDetailView, self).get_context_data(**kwargs)
-        # Create URL for packet download.
+        
+        # Get the event with prefetched media_urls in proper order. 
         event = context['event']
+        event = LAMetroEvent.objects.with_media().filter(ocd_id=event.ocd_id).first()
+        context['event'] = event
```
Comment (hancush) [inline lametro/views.py]: this looks like a good case for overriding [the `get_object` method](https://ccbv.co.uk/projects/Django/2.1/django.views.generic.detail/DetailView/#get_object)!

Comment (reginafcompton) [reply lametro/views.py]: GOOD IDEA

**Comment by reginafcompton** [comment]
THANK YOU HANNAH! YOU TOO

**Pull request merged**